pkg/data/format/format.go

```diff
@@ -0,0 +1,82 @@
+package format
+
+type ReferenceString interface {
+	Resolvable()
+	String() string
+}
+type OmittableField interface {
+	Omittable()
+}
+
+type Null interface {
+	Value
+}
+
+type Number interface {
+	Value
+	Integer() int
+	Float64() float64
+	String() string
+}
+
+type String interface {
+	Value
+	String() string
+}
+
+type Boolean interface {
+	Value
+	Boolean() bool
+	String() string
+}
+
+type ByteArray interface {
+	Value
+	ByteArray() []byte
+	String() string
+}
+
+type File interface {
+	Value
+
+	Binary() (ba ByteArray, err error)
+	DataURI() (url String, err error)
+	Base64() (url String, err error)
+	FileSize() (size Number)
+	ContentType() (t String)
+	FileName() (t String)
```

Comment:
```suggestion
	Filename() (t String)
```

Reply:
I’m considering this issue as well. I think both `Filename` and `File Name` are valid terms, with `Filename` being more commonly used in programming languages.

However, regarding the data attributes we’ve provided, we have many attributes prefixed with `File`, such as `File Size`, `File Name`, and `File Format`. Should we standardize their naming for consistency? This would result in using FileName or file-name, FileSize or file-size, and FileFormat or file-format, but not `Filename` and `filename`.

What do you think?